src/sys/stat.rs

```diff
@@ -402,12 +402,87 @@ pub fn lutimes<P: ?Sized + NixPath>(
     Errno::result(res).map(drop)
 }
 
+/// Timestamp update operation specifier used in [`utimensat`] and [`futimens`].
+#[cfg(not(target_os = "redox"))] // Redox does not support `UTIME_OMIT` and `UTIME_NOW`
+#[derive(Copy, Clone, Debug)]
+pub enum TimestampSpec {
+    /// Leave the timestamp unchanged.
+    Omit,
+    /// Update the timestamp to `Now`
+    Now,
+    /// Update the timestamp to the specific value specified in `TimeSpec`.
+    Set(TimeSpec),
+}
+
+/// A helper function to construct an instance of `libc::timespec` according to
+/// the operation specified in `spec`.
+#[cfg(not(target_os = "redox"))] // Redox does not support `UTIME_OMIT` and `UTIME_NOW`
+fn create_timespec(spec: &TimestampSpec) -> libc::timespec {
+    let mut default_value = libc::timespec {
+        tv_sec: 0,
+        tv_nsec: 0,
+    };
+
+    match spec {
+        #[cfg(all(
+            target_os = "linux",
+            target_arch = "x86_64",
+            target_pointer_width = "32"
+        ))]
+        TimestampSpec::Omit => default_value.tv_nsec = libc::UTIME_OMIT as i64,
+        #[cfg(not(all(
+            target_os = "linux",
+            target_arch = "x86_64",
+            target_pointer_width = "32"
+        )))]
+        TimestampSpec::Omit => default_value.tv_nsec = libc::UTIME_OMIT,
+        #[cfg(all(
+            target_os = "linux",
+            target_arch = "x86_64",
+            target_pointer_width = "32"
+        ))]
+        TimestampSpec::Now => default_value.tv_nsec = libc::UTIME_NOW as i64,
+        #[cfg(not(all(
+            target_os = "linux",
+            target_arch = "x86_64",
+            target_pointer_width = "32"
+        )))]
+        TimestampSpec::Now => default_value.tv_nsec = libc::UTIME_NOW,
```

Comment:
Let's deuglify this:
```suggestion
        TimestampSpec::Omit => default_value.tv_nsec = libc::UTIME_OMIT as _,
        TimestampSpec::Now => default_value.tv_nsec = libc::UTIME_NOW as _,
```

Reply:
Thanks for showing me this!